library/src/main/java/com/xfinity/blueprint/ViewAdapter.kt

```diff
@@ -0,0 +1,41 @@
+package com.xfinity.blueprint
+
+import android.view.View
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerView.SCROLL_STATE_IDLE
+import com.xfinity.blueprint.model.Component
+
+class ViewAdapter(private val recyclerView: RecyclerView,
+                  private var loadingView: View? = null) : ViewRegistry {
+
+    override fun updateComponents(newComponents: List<Component>, behavior: () -> Unit) {
+        if (!recyclerView.isComputingLayout) {
+            behavior.invoke()
+            return
+        }
+        recyclerView.addOnScrollListener(RecyclerViewScrollHandler(behavior))
+    }
+
+    override fun showLoading() {
+        loadingView?.visibility = View.VISIBLE
+    }
+
+    override fun hideLoading() {
+        loadingView?.visibility = View.GONE
+    }
```

Comment:
What's the benefit of having moving loading work here? Is screenviewdelegate delegating also delegating?

Reply:
To abstract away View from the Delegate.